src/develop/imageop.c

```diff
@@ -1363,56 +1363,57 @@ static void _set_trouble_message(dt_iop_module_t *const module, const char* cons
                                  const char* const trouble_tooltip, const char *const stderr_message)
 {
   GtkWidget *label_widget = NULL;
+  GtkWidget *iopw = NULL;
 
   if(module && module->widget)
   {
-    GtkWidget *iopw = gtk_widget_get_parent(module->widget);
+    iopw = gtk_widget_get_parent(module->widget);
     GList *children = gtk_container_get_children(GTK_CONTAINER(iopw));
     label_widget = g_list_nth_data(children, 0);
     g_list_free(children);
```

Comment:
Can you use module->warning_label to cache the widget pointer and avoid all of this work?  99.99+% of calls will have trouble_message==NULL and the module not currently in trouble.  And this function will be called multiple times per pixelpipe run.

Reply:
If the interface can be disambiguated so that `has_trouble == TRUE` is always in sync with `trouble_msg != NULL` then has_trouble can be used to determine if there is a widget that needs to be removed if there is no (new) trouble_msg. I'd rather clarify/simplify the interface than add cache fields to paper over the inefficiencies it creates.

Reply:
That would work as well.  Let me see whether merging set_module_in_trouble into set_module_trouble_message causes any problems elsewhere.

Reply:
Hi @ralfbrown were you still looking at this or was this completely resolved in another PR? I lost track a little, sorry.